src/build_id.cc

```diff
@@ -0,0 +1,25 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0. This product includes software
+// developed at Datadog (https://www.datadoghq.com/). Copyright 2021-Present
+// Datadog, Inc.
+
+#include "build_id.hpp"
+
+#include <iomanip>
+
+namespace ddprof {
+
+BuildIdStr format_build_id(BuildIdSpan build_id_span) {
+  std::string build_id_str;
+  // We encode every char on 2 hexa string characters
+  build_id_str.resize(build_id_span.size() * 2 + 1);
```

Comment:
Why `+1` ?

Reply:
snprintf adds an extra null termination

Reply:
Is it a problem? It would overwrite the existing null terminator of `build_id_str` and this is ok.
From https://en.cppreference.com/w/cpp/string/basic_string/operator_at:

> If pos == size(), a reference to the character with value CharT() (the null character) is returned.
For the first (non-const) version, the behavior is undefined if this character is modified to any value other than CharT() .